v2/client.go

```diff
@@ -96,9 +103,21 @@ func (c *Client) init() {
 		if c.UserAgent == "" {
 			c.UserAgent = defaultUserAgent
 		}
+		if c.OAuthConfig != nil {
+			oauthConfig := clientcredentials.Config{
+				ClientID:     c.OAuthConfig.ClientID,
+				ClientSecret: c.OAuthConfig.ClientSecret,
+				TokenURL:     c.BaseURL.ResolveReference(oauthRelTokenURL).String(),
+				Scopes:       c.OAuthConfig.Scopes,
+			}
+			// use context.Background() here, since this is used to refresh the token in the future
+			c.HTTP = oauthConfig.Client(context.Background())
+			c.HTTP.Timeout = defaultHttpClientTimeout
```

Comment:
HTTP, not Http per https://go.dev/wiki/CodeReviewComments#initialisms

Reply:
V1 carryover, but you're right.

Reply:
Fixed